modules/memory-management/src/main/java/org/mule/runtime/internal/memory/bytebuffer/ThreadLocalPool.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.internal.memory.bytebuffer;
+
+import java.nio.ByteBuffer;
+
+/**
+ * A thread local pool used by a {@link org.mule.runtime.api.memory.provider.ByteBufferProvider} to create and modify Buffers
+ *
+ * @param <T> Type of Buffer that will be created
+ *
+ *            Based on Grizzly Implementation
+ *
+ * @since 4.5.0
+ */
+public interface ThreadLocalPool<T extends ByteBuffer> {
```

Comment:
Can this be called ByteBufferPool or something like that? (ThreadLocalByteBufferPool can be the implementation)